R/carehomes.R

```diff
@@ -428,14 +436,15 @@ carehomes_initial <- function(info, n_particles, pars) {
   index_N_tot3 <- index[["N_tot3"]][[1L]]
 
   index_S <- index[["S"]]
```

Comment:
This is going to require some changes in `sircovid::carehomes_Rt_trajectories`, Anne suggests a weighted sum (relative to susceptibility/efficacy) over the S categories to give the same size S vector as before. Attention: @edknock

Reply:
@richfitz and @edknock I have now implemented changed in `sircovid::carehomes_Rt_trajectories` to allow for Rt calculation with the additional layer of stratification for S (now S has 2 dimensions, for age and vaccination status). 
I think this works but I've run into issues with tests which were using pre-recorded data for which the `rel_susceptibility` parameter was absent. I've artificially added it back with its default value of `1`, but that's pretty ugly. @richfitz if you can tell me how to better address this it would be great.
And @edknock if you could please check you agree with the changes I've made the the Rt calculation that would be much appreciated. 
Thanks!

Reply:
@richfitz see for example [this:](https://github.com/mrc-ide/sircovid/blob/5921e5f091746c8572f7165d7641df1967bca0b4/tests/testthat/test-carehomes-rt.R#L10)

Reply:
and @edknock please see [here](https://github.com/mrc-ide/sircovid/blob/5921e5f091746c8572f7165d7641df1967bca0b4/R/carehomes_rt.R#L37)